forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+class user_activity_search{
+    
+    private $searchArr;
+    private $dbResult;
+
+    public function match_request($request) 
+    {
+        $this->searchArr = $_POST;
+        if(!isset($_POST['condition']) || !isset($_POST['resultsCount'])){
+            header('Location: ../../');
+            exit;
+        }
+
+        if(qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR){
+            if(isset($this->searchArr)){
+                $this->dbSearch();
+                if(isset($this->dbResult)){
+                    return $request == 'user-activity-log-search';
+                }
+            }
+        }else{
+            return false;
+        }
+    }
+
+    public function process_request($request)
+    {
+        $qa_content=qa_content_prepare();
+        $qa_content['title']= qa_lang_html('user-activity-log/searchResults').$this->searchArr['request'];
+        
+        $qa_content['custom'] = $this->generateResultsTable();
+        $qa_content['navigation']['sub']=qa_admin_sub_navigation();
+        return $qa_content;
+    }
+
+    private function dbSearch()
+    {
+        $baseQuery = 'SELECT `datetime`, `ipaddress`, `handle`, `event` FROM qa_eventlog ';
+
+        if($this->searchArr['condition'] == 'username'){
+            $finalQuery = $baseQuery.'WHERE  handle = $';
+        }else if($this->searchArr['condition'] == 'type'){
+            $finalQuery = $baseQuery.'WHERE event = $';
+        }else if($this->searchArr['condition'] == 'ip'){
+            if(qa_get_logged_in_level() > QA_USER_LEVEL_EDITOR){
+                $finalQuery = $baseQuery.'WHERE ipaddress = $';
+            }   
+        }
+
+        if($this->searchArr['date'] != ""){
+            $dateArr = explode('-', $this->searchArr['date']);
+
+
+            if(is_numeric($dateArr[0]) && is_numeric($dateArr[0])){
+                $date = $dateArr[0];
+                if(isset($dateArr[1]) && is_numeric($dateArr[1])){
+                    $date = $date.'-'.$dateArr[1];
+                    if(isset($dateArr[2]) && is_numeric($dateArr[2])){
+                        if(strlen($dateArr[2]) > 2){
+                            $time = explode(' ', $dateArr[2]);
```

Comment:
Tu chyba można użyć [destrukturyzacji](https://www.php.net/manual/en/migration71.new-features.php#migration71.new-features.symmetric-array-destructuring)?

Reply:
Jeżeli dobrze rozumiem ten mechanizm (nigdy wcześniej z niego nie korzystałem), potrzebujemy w niej wiedzieć ile będzie pól tablicy. Tutaj niezbyt wiemy ponieważ możemy wyszukiwać dowolnie tzn.

możemy wpisać
2022
2022-02
2022-02-01
2022-02-01 09
etc.
i każda z tych dat da nam prawidłowy wynik

Możemy dostać tych pól w tablicy max 6 ale jeśli przekażemy mniej dostaniemy błąd ( z tego co zdążyłem przetestować)

Reply:
```php
$time = explode(' ', $dateArr[2]);

$dateArr[2] = $time[0];
$dateArr[3] = $time[1];
```

Skoro bierzesz pola z `$time[0]` i `$time[1]`, to z góry wiesz jakie indeksy/propertisy chcesz użyć, więc destrukturyzacja powinna tutaj zadziałać (kwestia tylko nazwać odpowiednio rozpakowane zmienne zgodnie z tym, co reprezentują).
```php
[$firstTime, $secondTime] = explode(' ', $dateArr[2]);

$dateArr[2] = $firstTime;
$dateArr[3] = $secondTime;
```